packages/entity-browser/src/components/DetailView/DetailForm.js

```diff
@@ -33,16 +33,20 @@ export const DetailForm = props => {
           let fieldProps = {}
           if (entityField.type === 'field') {
             fieldProps.type = entityField.value.type
-          } else if (entityField.type === 'entity') {
+          } else if (entityField.type === 'entity' && props.stores[field.name] !== undefined) {
             fieldProps.type = 'single-select'
-            fieldProps.options = {store: [{value: entityField.value.key, label: entityField.value.display}]}
-          } else if (entityField.type === 'entity-list') {
+            const store = props.stores[field.name].data
+            fieldProps.options = {store}
+          } else if (entityField.type === 'entity-list' && props.stores[field.name] !== undefined) {
             fieldProps.type = 'multi-select'
-            const store = entityField.value.map(e => ({value: e.key, label: e.display}))
+            const store = props.stores[field.name].data
             fieldProps.options = {store}
+          } else {
+            fieldProps.options = {store: []}
```

Comment:
i get why you have this `else` but its not really a default fallback (to an unknown type). better remove it and change line 38 / 42 to this:

  const store = props.stores[field.name] ? props.stores[field.name].data : []